sdk/communication/AzureCommunicationChat/Source/ChatClient.swift

```diff
@@ -0,0 +1,182 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import AzureCommunication
+import AzureCore
+import Foundation
+
+public class ChatClient {
+    // MARK: Properties
+
+    private let endpoint: String
+    private let credential: CommunicationUserCredential
+    private let options: AzureCommunicationChatClientOptions
+    private let service: AzureCommunicationChatService
+
+    // MARK: Initializers
+
+    /// Create a ChatClient.
+    /// - Parameters:
+    ///   - endpoint: The Communication Services endpoint.
+    ///   - credential: The user credential.
+    ///   - options: Options used to configure the client.
+    public init(
+        endpoint: String,
+        credential: CommunicationUserCredential,
+        withOptions options: AzureCommunicationChatClientOptions
+    ) throws {
+        self.endpoint = endpoint
+        self.credential = credential
+        self.options = options
+
+        guard let endpointUrl = URL(string: endpoint) else {
+            throw AzureError.client("Unable to form base URL.")
+        }
+
+        let authPolicy = CommunicationUserCredentialPolicy(credential: credential)
+
+        let client = try AzureCommunicationChatClient(
+            endpoint: endpointUrl,
+            authPolicy: authPolicy,
+            withOptions: options
+        )
+
+        self.service = client.azureCommunicationChatService
+    }
+
+    // MARK: Public Methods
+
+    /// Create a ChatThreadClient for the ChatThread with id threadId.
+    /// - Parameters:
+    ///   - forChatThread: The threadId.
```

Comment:
If you use XCode's built-in documentation function, this would be populated `threadId`. You should follow that convention.